src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WebSearch/WebSearchTopLevelCommandItem.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.CmdPal.Ext.WebSearch.Commands;
+using Microsoft.CmdPal.Ext.WebSearch.Helpers;
+using Microsoft.CmdPal.Ext.WebSearch.Pages;
+using Microsoft.CmdPal.Ext.WebSearch.Properties;
+using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
+
+namespace Microsoft.CmdPal.Ext.WebSearch;
+
+public partial class WebSearchTopLevelCommandItem : CommandItem, IFallbackHandler
+{
+    private readonly SettingsManager _settingsManager;
+
+    public WebSearchTopLevelCommandItem(SettingsManager settingsManager)
+        : base(new WebSearchListPage(settingsManager))
+    {
+        Icon = new(Path.Combine(AppDomain.CurrentDomain.BaseDirectory.ToString(), "Images\\WebSearch.dark.png"));
+        SetDefaultTitle();
+        _settingsManager = settingsManager;
+    }
+
+    private void SetDefaultTitle() => Title = Resources.command_item_title;
+
+    public void UpdateQuery(string query)
+    {
+        if (string.IsNullOrEmpty(query))
+        {
+            SetDefaultTitle();
+            Command = new WebSearchListPage(_settingsManager);
+        }
+        else
+        {
+            Title = query;
+            var searchArgs = $"? {query}";
```

Comment:
I think this _feels_ weird to me. We're like, manually prepending `? ` to the search text, just to trim it off again inside `OpenCommandInShell`?

Reply:
Changed it so that the `$"? {stuff}"` happens only inside the `WebSearchCommand` so it is more clean.